src/utils/collections/iMap.ml

```diff
@@ -3,7 +3,7 @@
  * All rights reserved.
  *
  * This source code is licensed under the BSD-style license found in the
```

Comment:
you've grabbed a very old version of some of these files - if you go look at facebook/hhvm again, you'll find the most recent versions are MIT-licensed

Reply:
Yep--it's worth noting that these versions are consistent with the ones in [Pyre](https://github.com/facebook/pyre-check/blob/master/hack_parallel/hack_parallel/utils/collections/iMap.ml), and that's where I took it from. It's worth upstreaming the current files in Hack to Pyre then (and seeing if it breaks anything there). I'm worried that if I just drop in the current Hack files from master that it'll break the interface. Happily accepting PRs; for now it's low priority to check that these are current with Hack. Tracking in #4.